### DISCUSSION TITLE
该如何理解 路由里面的布局方式？

### DISCUSSION THREAD

**Discussion by zuihou** [soybeanjs/soybean-admin#334] [Q&A]
`layout.base`、`layout.blank`、`layout.base$view.xxx`、`layout.blank$view.xxx` 分别是什么意思呢？

在什么场景该使用什么？

**Comment by honghuangdc** [comment]
可以看下插件[ElegantRouter](https://github.com/soybeanjs/elegant-router)文档

https://github.com/soybeanjs/elegant-router/blob/main/README.zh_CN.md#%E4%B8%80%E7%BA%A7%E8%B7%AF%E7%94%B1%E5%8D%95%E7%BA%A7%E8%B7%AF%E7%94%B1

**Reply by zuihou** [reply]
看了几遍了，没太懂啊。
关键是没搞懂 `layouts/blank-layout/index.vue` 和 `layouts/base-layout/index.vue` 这2个的区别。

**Reply by honghuangdc** [reply]
不同的布局采用不同的文件

**Reply by zuihou** [reply]
这2个布局有什么区别？  

blank-layout 表示全屏页面？   比如登录页面？
base-layout 表示左边是菜单，顶部是导航栏？  比如用户管理、角色管理？

**Reply by zuihou** [reply]
另外，
1. 是不是这有1级菜单( isSingleLevelRoute(route) ) 设置 layout.base$view.xxx、layout.blank$view.xxx  才有用？ 2级菜单设置layout.xxx 没用，只能设置view.xxx ?
2. “目录” 只能设置为 layout.base 。 例如这些目录
<img width="298" alt="image" src="https://github.com/soybeanjs/soybean-admin/assets/8500410/92acb08e-68b1-4148-9ffb-5d6d3e2c8b91">

**Comment by zuihou** [comment]
另外：
1. 是否支持  /user/:type/:id  这种路径
2. props 属性是什么意思

<img width="613" alt="image" src="https://github.com/soybeanjs/soybean-admin/assets/8500410/99269d3c-04d3-43ef-aa9b-ee3a4cd43465">

**Reply by honghuangdc** [reply]
props为true是vue-router里面的，将参数转换为组件属性

**Comment by zuihou** [comment]
<img width="496" alt="image" src="https://github.com/soybeanjs/soybean-admin/assets/8500410/510c4175-a393-46f0-abd8-9f17fa7d3f67">

”忽略文件夹的聚合路由“ 能否支持配置为指定文件夹下的都忽略？  因为历史项目写了很多页面了，在迁移为本框架时，希望保留原来的代码路径。

**Reply by honghuangdc** [reply]
这里的忽略聚合是只生成的路由名称不包含以_开头的名称，如果要想不检测指定文件夹下的文件，请看文档，里面有个pageExcludePatterns属性

**Comment by honghuangdc** [comment]
> <img width="496" alt="image" src="https://github.com/soybeanjs/soybean-admin/assets/8500410/510c4175-a393-46f0-abd8-9f17fa7d3f67">
> 
> 
> 
> ”忽略文件夹的聚合路由“ 能否支持配置为指定文件夹下的都忽略？  因为历史项目写了很多页面了，在迁移为本框架时，希望保留原来的代码路径。

需要帮助的话，可以加一下我的微信 soybeanjs 拉你入群，有问题可以在群里面提问，或者找我提供有偿协助
